assembly/src/parsers/ast/context.rs

```diff
@@ -255,10 +255,17 @@ impl ParserContext {
         &self,
         tokens: &mut TokenStream,
         nodes: &mut Vec<Node>,
+        if_true_branch: bool,
```

Comment:
nit: I'd probably rename this into something like `break_on_else` to make the meaning a bit more clear.